src/crypto/libcrypto.c

```diff
@@ -35,26 +35,26 @@
 
 #if OPENSSL_VERSION_NUMBER < 0x10100000L || \
    (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x20700000L)
-EVP_CIPHER_CTX *
+static EVP_CIPHER_CTX *
```

Comment:
This is producing a warning on several tasks in CI:

```
[2022/02/16 17:32:23.583] [ 13%] Building C object CMakeFiles/mongocrypt_static.dir/src/crypto/libcrypto.c.o
[2022/02/16 17:32:23.633] /data/mci/aed13a8302333f65733d1a328260992e/libmongocrypt/src/crypto/libcrypto.c:39:1: error: static declaration of 'EVP_CIPHER_CTX_new' follows non-static declaration
[2022/02/16 17:32:23.633]  EVP_CIPHER_CTX_new (void)
[2022/02/16 17:32:23.633]  ^~~~~~~~~~~~~~~~~~
[2022/02/16 17:32:23.633] In file included from /data/mci/aed13a8302333f65733d1a328260992e/libmongocrypt/src/crypto/libcrypto.c:31:0:
[2022/02/16 17:32:23.633] /usr/include/openssl/evp.h:695:17: note: previous declaration of 'EVP_CIPHER_CTX_new' was here
[2022/02/16 17:32:23.633]  EVP_CIPHER_CTX *EVP_CIPHER_CTX_new(void);
[2022/02/16 17:32:23.633]                  ^~~~~~~~~~~~~~~~~~
[2022/02/16 17:32:23.633] /data/mci/aed13a8302333f65733d1a328260992e/libmongocrypt/src/crypto/libcrypto.c:45:1: error: static declaration of 'EVP_CIPHER_CTX_free' follows non-static declaration
[2022/02/16 17:32:23.633]  EVP_CIPHER_CTX_free (EVP_CIPHER_CTX *ctx)
[2022/02/16 17:32:23.633]  ^~~~~~~~~~~~~~~~~~~
[2022/02/16 17:32:23.633] In file included from /data/mci/aed13a8302333f65733d1a328260992e/libmongocrypt/src/crypto/libcrypto.c:31:0:
[2022/02/16 17:32:23.633] /usr/include/openssl/evp.h:696:6: note: previous declaration of 'EVP_CIPHER_CTX_free' was here
[2022/02/16 17:32:23.633]  void EVP_CIPHER_CTX_free(EVP_CIPHER_CTX *a);
[2022/02/16 17:32:23.633]       ^~~~~~~~~~~~~~~~~~~
[2022/02/16 17:32:23.638] gmake[2]: *** [CMakeFiles/mongocrypt_static.dir/src/crypto/libcrypto.c.o] Error 1
[2022/02/16 17:32:23.638] gmake[1]: *** [CMakeFiles/mongocrypt_static.dir/all] Error 2
[2022/02/16 17:32:23.638] gmake: *** [all] Error 2
```

I will have to investigate.

Reply:
Seems that used version of openssl provides these functions, but is not catched by the condition...

Reply:
From a quick check:
* `EVP_CIPHER_CTX_new` and `EVP_CIPHER_CTX_free` are available in ALL version of LibreSSL (checked in 2.0.0)
* `HMAC_CTX_new` and `HMAC_CTX_free` are available in LibreSSL since 3.0.0.

Reply:
`EVP_CIPHER_CTX_new` and `EVP_CIPHER_CTX_free` are available, at least in OpenSSL 1.0.2  (RHEL-7), so perhaps these 2 polyfills can be removed.